src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java

```diff
@@ -0,0 +1,390 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.mercury;
+
+import java.util.List;
+
+import io.appium.java_client.MultiTouchAction;
+import io.appium.java_client.TouchAction;
+import io.appium.java_client.android.AndroidDriver;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import com.wikia.webdriver.common.driverprovider.NewDriverProvider;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+/**
+ * @authors: Tomasz Napieralski
+ * @created: 9 Jan 2015
+ * @updated: 28 Jan 2015
+ */
+
+public class PerformTouchAction {
+	
+	private WebDriver driver;
+	private AndroidDriver mobileDriver;
+	
+	private int nativeHeight = 0;
+	private int nativeWidth = 0;
+	private int webviewHeight = 0;
+	private int webviewWidth = 0;
+	private int ratio = 0;
+	
+	private int taskbarNativeHeight = 0;
+	private int taskbarNativeWidth = 0;
+	private int taskbarWebviewHeight = 0;
+	private int taskbarWebviewWidth = 0;
+	
+	private int appNativeHeight = 0;
+	private int appNativeWidth = 0;
+	private int appWebviewHeight = 0;
+	private int appWebviewWidth = 0;
+	
+	private int loadedPageHeight = 0;
+	private int loadedPageWidth = 0;
+	
+	public static final String DIRECTION_LEFT = "left";
+	public static final String DIRECTION_RIGHT = "right";
+	public static final String DIRECTION_UP = "up";
+	public static final String DIRECTION_DOWN = "down";
+	
+	public static final String ZOOM_WAY_IN = "in";
+	public static final String ZOOM_WAY_OUT = "out";
+	
+	public PerformTouchAction(WebDriver webDriver) {
+		mobileDriver = NewDriverProvider.getMobileDriver();
+		driver = webDriver;
+		JavascriptExecutor js = (JavascriptExecutor) driver;
+		if (!"NATIVE_APP".equals(mobileDriver.getContext())) {
+			mobileDriver.context("NATIVE_APP");
+		}
+		nativeHeight = mobileDriver.manage().window().getSize().height;
+		nativeWidth = mobileDriver.manage().window().getSize().width;
+		int startX = nativeWidth / 2;
+		int startY = nativeHeight - 1;
+		int endX = startX;
+		int endY = nativeHeight / 3;
+		int duration = startY - endY;
+		try {
+			mobileDriver.swipe(startX, startY, endX, endY, duration);
+		} catch (Exception e) {
+			PageObjectLogging.log("PerformTouchAction", e.toString(), false);
+		}
+		try {
+			Thread.sleep(duration * 3);
+		} catch(Exception e) {
```

Comment:
Please specified which exception the try block is catching rather than a generic exception.